gitbook-docs/rest_api.md

```diff
@@ -65,3 +65,9 @@ of the data under the `vessels` container in JSON format. Likewise, `GET
 /signalk/v1/api/vessels/urn:mrn:signalk:uuid:c0d79334-4e25-4245-8892-54e8ccc8021d` should return data for one specific
 vessel. In other words, the full Signal K data model SHOULD be traversable by any client making GET requests to an
 arbitrary depth.
+
+## History snapshot retrieval
+
+A server MAY support retrieving historical data. The client specifies the requested timestamp with request parameter `time`, for example `https://localhost:3443/signalk/v1/api/vessels/self?time=2018-08-24T15:19:09Z`. The server will attempt to create the request part of the full model at the requested time.
+
+Since this is an addition to the 1.0 Signal K the client can not reliably detect if the server supports history snapshot retrieval from http status code, but should check the timestamps in the returned data. A server MAY respond with `501 Not Implemented` status code if it does not support history snapshot retrieval and with `400 Bad Request` if it does not have data for the requested timestamp.
```

Comment:
Why don't we make this a 1.1 feature?

Reply:
Another solution is to add `/signalk/v1/snapshot`, which has the advantage of being naturally 404 when unimplemented. If `time` is supported on `/signalk/v1/api` we just redirect. So its easy to check if its supported, and if so you can just use the std api?

Or we could start a `/signalk/v1.1/api` that included the new functionality. Trouble is we will get a zillion `/v1.x/` variants as we add to it, so it wont be nice.

Reply:
In fact if we make a `/signalk/v1/feature/xxx` structure, we can add features as we please. The API users should then code so they test `/signak/v1/feature/xxx` before using a feature.

Reply:
Do we really need the `/feature/` part in the path - why not just `/signalk/v1/snapshot` like you suggested?

Reply:
I meant that if we dont use a separate `/signalk/v1/snapshot` path then we could use a `/signalk/v1/feature/snapshot` url to test if the server has the snapshot capability. It would 404 for a server version that did not support it, so the client would know not to try `/signalk/v1/api?time-=...` as it would just return now(). Moot if we use separate paths which is cleaner anyway.